test/dynamics/pulse/test_string_model_parser.py

```diff
@@ -0,0 +1,422 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+# pylint: disable=invalid-name
+
+"""
+Tests for pulse.string_model_parser for importing qiskit.pulse model string representation.
+"""
+
+import numpy as np
+
+from qiskit import QiskitError
+from qiskit.quantum_info.operators import Operator
+
+from qiskit_dynamics.pulse import parse_hamiltonian_dict
+from qiskit_dynamics.pulse.string_model_parser.string_model_parser import (
+    hamiltonian_pre_parse_exceptions,
+)
+
+from qiskit_dynamics.type_utils import to_array
+
+from ..common import QiskitDynamicsTestCase
+
+
+class TestHamiltonianPreParseExceptions(QiskitDynamicsTestCase):
+    """Tests for preparse formatting exceptions."""
+
+    def test_no_h_str(self):
+        """Test no h_str empty raises error."""
+
+        with self.assertRaises(QiskitError) as qe:
+            hamiltonian_pre_parse_exceptions({})
+        self.assertTrue("requires a non-empty 'h_str'" in str(qe.exception))
+
+    def test_empty_qub(self):
+        """Test qub dict empty raises error."""
+
+        with self.assertRaises(QiskitError) as qe:
+            hamiltonian_pre_parse_exceptions({"h_str": ["a * X0|||D0"]})
+        self.assertTrue("requires non-empty 'qub'" in str(qe.exception))
+
+    def test_too_many_vertical_bars(self):
```

Comment:
Add a test for too few vertical bars (1)

Reply:
Added below this one.